Source/SwiftLintFramework/Rules/Style/MultipleClosuresWithTrailingClosureRule.swift

```diff
@@ -38,101 +37,34 @@ public struct MultipleClosuresWithTrailingClosureRule: ASTRule, ConfigurationPro
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
     }
 }
 
-private extension SourceKittenDictionary {
-    func closingParenLocation(file: SwiftLintFile) -> ByteCount? {
-        guard self.expressionKind == .call,
-              case let arguments = self.enclosedArguments,
-              arguments.isNotEmpty else {
-            return nil
-        }
+private extension MultipleClosuresWithTrailingClosureRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
 
-        func rangeBetween(_ expr1: SourceKittenDictionary, and expr2: SourceKittenDictionary) -> ByteRange? {
-            guard let offset1 = expr1.offset,
-                  let length1 = expr1.length,
-                  let offset2 = expr2.offset,
-                  case let end1 = offset1 + length1,
-                  end1 <= offset2 else {
-                return nil
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            guard let trailingClosure = node.trailingClosure,
+                  node.hasMultipleClosuresWithTrailingClosureViolation else {
+                return
             }
 
-            return ByteRange(location: end1, length: offset2 - end1)
+            violationPositions.append(trailingClosure.positionAfterSkippingLeadingTrivia)
         }
-
-        var searchRanges: [ByteRange] = []
-        for index in arguments.indices.dropLast() {
-            let currentArg = arguments[index]
-            let nextArg = arguments[index + 1]
-            if let range = rangeBetween(currentArg, and: nextArg) {
-                searchRanges.append(range)
-            }
-        }
-
-        if let lastOffset = arguments.last?.offset,
-           let lastLength = arguments.last?.length,
-           let callOffset = self.offset,
-           let callLength = self.length,
-           case let lastEnd = lastOffset + lastLength,
-           case let callEnd = callOffset + callLength,
-           lastEnd <= callEnd {
-            searchRanges.append(ByteRange(location: lastEnd, length: callEnd - lastEnd))
-        }
-
-        for byteRange in searchRanges {
-            if let range = file.stringView.byteRangeToNSRange(byteRange),
-               let match = regex("^\\s*\\)").firstMatch(in: file.contents, options: [], range: range)?.range {
-                return file.stringView.byteOffset(fromLocation: match.location)
-            }
-        }
-
-        return nil
     }
 }
 
-private extension Array where Element == SourceKittenDictionary {
-    func filterClosures(file: SwiftLintFile) -> [SourceKittenDictionary] {
-        return filter { argument in
-            return argument.substructure.contains(where: { dictionary in
-                dictionary.expressionKind == .closure
-            })
+private extension FunctionCallExprSyntax {
+    var hasMultipleClosuresWithTrailingClosureViolation: Bool {
```

Comment:
Maybe just `hasTrailingClosureViolation` for terseness and to satisfy the variable name rule?

```suggestion
    var hasTrailingClosureViolation: Bool {
```